src/components/output/index.tsx

```diff
@@ -10,34 +14,88 @@ type State = {};
 
 export default class App extends Component<Props, State> {
   state: State = {};
-  canvas?: HTMLCanvasElement;
+  canvasLeft?: HTMLCanvasElement;
+  canvasRight?: HTMLCanvasElement;
+  pinchZoomLeft?: PinchZoom;
+  pinchZoomRight?: PinchZoom;
+  retargetedEvents = new WeakSet<Event>();
 
-  constructor() {
-    super();
-  }
-
-  updateCanvas(img: ImageBitmap) {
-    if (!this.canvas) return;
-    const ctx = this.canvas.getContext('2d');
-    if (!ctx) return;
-    ctx.clearRect(0, 0, this.canvas.width, this.canvas.height);
-    ctx.drawImage(img, 0, 0);
+  updateCanvases(img: ImageBitmap) {
+    for (const [i, canvas] of [this.canvasLeft, this.canvasRight].entries()) {
+      if (!canvas) return;
+      const ctx = canvas.getContext('2d');
+      if (!ctx) return;
+      if (i === 1) {
+        // This is temporary, to show the images are different
+        ctx.filter = 'hue-rotate(180deg)';
+      }
+      ctx.clearRect(0, 0, canvas.width, canvas.height);
+      ctx.drawImage(img, 0, 0);
+    }
   }
 
   componentDidMount() {
-    this.updateCanvas(this.props.img);
+    this.updateCanvases(this.props.img);
   }
 
   componentDidUpdate({ img }: Props) {
-    if (img !== this.props.img) this.updateCanvas(this.props.img);
+    if (img !== this.props.img) this.updateCanvases(this.props.img);
+  }
+
+  @bind
+  onPinchZoomLeftChange(event: Event) {
+    if (!this.pinchZoomRight || !this.pinchZoomLeft) return;
+    this.pinchZoomRight.setTransform({
+      scale: this.pinchZoomLeft.scale,
+      x: this.pinchZoomLeft.x,
+      y: this.pinchZoomLeft.y
+    });
+  }
+
+  /**
+   * We're using two pinch zoom elements, but we want them to stay in sync. When one moves, we
+   * update the position of the other. However, this is tricky when it comes to multi-touch, when
+   * one finger is on one pinch-zoom, and the other finger is on the other. To overcome this, we
+   * redirect all relevant pointer/touch/mouse events to the first pinch zoom element.
+   *
+   * @param event Event to redirect
+   */
+  @bind
+  onRetargetableEvent(event: Event) {
+    const targetEl = event.target as HTMLElement;
+    if (!this.pinchZoomLeft) return;
+    // If the event is on the handle of the two-up, let it through.
+    if (targetEl.closest('.' + twoUpHandle)) return;
+    // If we've already retargeted this event, let it through.
+    if (this.retargetedEvents.has(event)) return;
+    // Stop the event in its tracks.
+    event.stopImmediatePropagation();
+    event.preventDefault();
+    // Clone the event & dispatch
+    const clonedEvent = new (event.constructor as typeof Event)(event.type, event);
```

Comment:
Surprised you need to force the type of `event.constructor`, but I’m guessing you did try without.

Reply:
I should add a comment linking to https://github.com/Microsoft/TypeScript/issues/3841.